lib/ddtrace/contrib/active_support/notifications/event.rb

```diff
@@ -56,6 +56,40 @@ def tracer
             end
           end
         end
+
+        # Extension to {Event} class that ensures the current {Context}
+        # is always clean when the event is processed.
+        #
+        # This is a safeguard as Contexts are thread-bound.
+        # If an integration re-uses threads, the context from a previous
+        # execution could leak into the new execution.
+        #
+        # This module *cannot* be used for events can be nested, as
+        # it drops all spans currently active in the {Context}.
+        module RootEvent
+          def subscription(*args)
+            super.tap do |subscription|
+              subscription.before_trace { ensure_clean_context! }
+            end
+          end
+
+          private
+
+          # Clears context if there are unfinished spans in it
+          def ensure_clean_context!
```

Comment:
I like the idea of extracting this function out and standardizing this here.

Does it make a sense to go a step further and extract this function to a module available to all integrations? Other integrations do the same, such as Rack and SuckerPunch, where they reset the context as well. If we're standardizing here, I don't think its a stretch to standardize everywhere, if plausible.